phpBB/feed.php

```diff
@@ -23,222 +26,32 @@
 $phpbb_root_path = (defined('PHPBB_ROOT_PATH')) ? PHPBB_ROOT_PATH : './';
 $phpEx = substr(strrchr(__FILE__, '.'), 1);
 include($phpbb_root_path . 'common.' . $phpEx);
-include($phpbb_root_path . 'includes/functions_display.' . $phpEx);
 
-if (!$config['feed_enable'])
-{
-	trigger_error('NO_FEED_ENABLED');
-}
+/** @var \phpbb\controller\helper $controller_helper */
+$controller_helper = $phpbb_container->get('controller.helper');
 
-// Start session
-$user->session_begin();
-
-if (!empty($config['feed_http_auth']) && $request->variable('auth', '') == 'http')
-{
-	phpbb_http_login(array(
-		'auth_message'	=> 'Feed',
-		'viewonline'	=> $request->variable('viewonline', true),
-	));
-}
-
-$auth->acl($user->data);
-$user->setup('viewtopic');
-
-// Initial var setup
 $forum_id	= $request->variable('f', 0);
 $topic_id	= $request->variable('t', 0);
-$mode		= $request->variable('mode', '');
-
-// We do not use a template, therefore we simply define the global template variables here
-$global_vars = $item_vars = array();
-$feed_updated_time = 0;
+$mode	= $request->variable('mode', '');
 
-// Generate params array for use in append_sid() to correctly link back to this page
-$params = false;
-if ($forum_id || $topic_id || $mode)
+if ($forum_id !== 0)
 {
-	$params = array(
-		'f'		=> ($forum_id) ? $forum_id : NULL,
-		't'		=> ($topic_id) ? $topic_id : NULL,
-		'mode'	=> ($mode) ? $mode : NULL,
-	);
+	$url = $controller_helper->route('phpbb_feed_forum', array('forum_id' => $forum_id));
 }
-
-// This boards URL
-/* @var $phpbb_feed_helper \phpbb\feed\helper */
-$phpbb_feed_helper = $phpbb_container->get('feed.helper');
-$board_url = $phpbb_feed_helper->get_board_url();
-
-// Get correct feed object
-/* @var $phpbb_feed_factory \phpbb\feed\factory */
-$phpbb_feed_factory = $phpbb_container->get('feed.factory');
-$feed = $phpbb_feed_factory->get_feed($mode, $forum_id, $topic_id);
-
-// No feed found
-if ($feed === false)
+else if ($topic_id !== 0)
 {
-	trigger_error('NO_FEED');
+	$url = $controller_helper->route('phpbb_feed_topic', array('topic_id' => $topic_id));
 }
-
-// Open Feed
-$feed->open();
-
-// Iterate through items
-while ($row = $feed->get_item())
+else
 {
-	// BBCode options to correctly disable urls, smilies, bbcode...
-	if ($feed->get('options') === NULL)
+	try
 	{
-		// Allow all combinations
-		$options = 7;
-
-		if ($feed->get('enable_bbcode') !== NULL && $feed->get('enable_smilies') !== NULL && $feed->get('enable_magic_url') !== NULL)
-		{
-			$options = (($row[$feed->get('enable_bbcode')]) ? OPTION_FLAG_BBCODE : 0) + (($row[$feed->get('enable_smilies')]) ? OPTION_FLAG_SMILIES : 0) + (($row[$feed->get('enable_magic_url')]) ? OPTION_FLAG_LINKS : 0);
-		}
+		$url = $controller_helper->route('phpbb_feed_overall', array('mode' => $mode));
 	}
-	else
-	{
-		$options = $row[$feed->get('options')];
+	catch (InvalidParameterException $e) {
```

Comment:
I think the PR for extending sniffer functionality to also recognize wrong braces was merged later, right? ;)